### PR TITLE
Bug fixes for analytics and translations for 2.0

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

* added null checks and error handling
* fixed link on community page
* fixed agent terminology
* fixed UI bugs on application dashboards and workflow comparison table

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

File: litmus-portal/frontend/public/locales/en/translation.yaml [hunk 1]
```diff
@@ -91,7 +91,7 @@ workflowStepper:
   back: Back
 
 editor:
-  status: 'YAML Status:'
+  status: "YAML Status:"
```
Comment (whitetiger1399) [inline litmus-portal/frontend/public/locales/en/translation.yaml:94]: May you revert the quotes changes are not required. They might have been added automatically on save. Please prefer reverting it.

Comment (whitetiger1399) [reply litmus-portal/frontend/public/locales/en/translation.yaml:94]: Same for all similar changes

File: litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx [hunk 2]
```diff
@@ -854,7 +860,7 @@ const DashboardPage: React.FC = () => {
                         }}
                         panel_group_id={panelGroup.panel_group_id}
                         panel_group_name={panelGroup.panel_group_name}
-                        panels={panelGroup.panels}
+                        panels={panelGroup.panels ?? panelGroup.panels}
```
Comment (whitetiger1399) [inline litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx:863]: May we revert this change. It does not add any advantage.

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx:863]: Sometimes the UI crashes if panels are undefined due to some illegal character via translation this this check has been added, will add a ternary operator instead.

File: litmus-portal/frontend/public/locales/en/translation.yaml (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/public/locales/en/translation.yaml:94]: Sure

File: litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/PanelAndGroup/Panel.tsx [hunk 3]
```diff
@@ -113,17 +113,20 @@ const DashboardPanel: React.FC<GraphPanelProps> = ({
...
           }
-          height="95% !important"
+          height="95%"
           width="95%"
         >
           <div className={classes.popOutModal}>
             <Typography className={classes.title}>{panel_name}</Typography>
             <LineAreaGraph
-              legendTableHeight={120}
+              legendTableHeight={150}
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/PanelAndGroup/Panel.tsx:129]: Yes I did check that and found some issues thus have updated.

[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/PanelAndGroup/Panel.tsx:129]: The popout mode in figma does not have a lot of empty space at bottom and ratio of graph to legend table is also a bit large as per current version.

File: litmus-portal/frontend/src/views/Analytics/ApplicationDashboards/Steps/TuneTheQueries/EditPanelsWizard/QueryEditingWizard/QueryEditor/index.tsx [hunk 4]
```diff
@@ -650,6 +712,7 @@ const QueryEditor: React.FC<QueryEditorProps> = ({
                   className={classes.formControl}
                   style={{ width: '8.5rem' }}
                   color="primary"
+                  disabled
```
Comment (whitetiger1399) [inline litmus-portal/frontend/src/views/Analytics/ApplicationDashboards/Steps/TuneTheQueries/EditPanelsWizard/QueryEditingWizard/QueryEditor/index.tsx:715]: Are we disabling the Resolution feature?

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Analytics/ApplicationDashboards/Steps/TuneTheQueries/EditPanelsWizard/QueryEditingWizard/QueryEditor/index.tsx:715]: Yes because currently the graphs don’t support changing resolution. It’s not required for the Prometheus queries.

File: litmus-portal/frontend/src/views/Analytics/WorkflowDashboard/WorkflowComparisonTable/index.tsx [hunk 5]
```diff
@@ -286,7 +286,7 @@ const WorkflowComparisonTable = () => {
               console.error(error);
             }
           });
-          if (isWorkflowValid) {
+          if (isWorkflowValid && selectedRuns.length) {
```
Comment (whitetiger1399) [inline litmus-portal/frontend/src/views/Analytics/WorkflowDashboard/WorkflowComparisonTable/index.tsx:289]: For a recurring workflow, may we show an empty graph or inform the user that there is not data

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Analytics/WorkflowDashboard/WorkflowComparisonTable/index.tsx:289]: Yes we are doing that, there is already a message as no runs available. @whitetiger1399

File: litmus-portal/frontend/src/views/Community/CommunityCards/index.tsx [hunk 6]
```diff
@@ -72,16 +72,8 @@ const CommunityCards: React.FC = () => {
...
-          window.open(
-            'https://www.meetup.com/Kubernetes-Chaos-Engineering-Meetup-Group'
-          )
-        }
-        onKeyDown={() =>
-          window.open(
-            'https://www.meetup.com/Kubernetes-Chaos-Engineering-Meetup-Group'
-          )
-        }
+        onClick={() => window.open('https://litmuschaos.io/community')}
```
Comment (whitetiger1399) [inline litmus-portal/frontend/src/views/Community/CommunityCards/index.tsx:75]: Please confirm this change with the Community Manager. @prithvi1307

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Community/CommunityCards/index.tsx:75]: Sure, this is as per discussion with @ajeshbaby but I will ask @prithvi1307 and @ksatchit to verify the same.

File: litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx (hunk 2 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx:863]: Please suggest if you have any better workarounds ? @whitetiger1399

**Comment by ishangupta-ds** [comment]
@whitetiger1399 suggested changes have been made PTAL again.

File: litmus-portal/frontend/src/views/Community/CommunityCards/index.tsx (hunk 6 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Community/CommunityCards/index.tsx:75]: Verified by @prithvi1307

File: litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx (hunk 2 above)
Comment (whitetiger1399) [reply litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx]: May replace it with `{ panelGroup.panels ?? [] }`

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx]: Done with previous commit.

**Review by whitetiger1399 — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]
